lib/ddtrace/contrib/rails/action_view.rb

```diff
@@ -34,17 +34,39 @@ def self.get_key(f)
           'datadog_actionview_' + f
         end
 
-        def self.start_render_template(*)
-          key = get_key('render_template')
-          return if Thread.current[key]
+        def self.start_render_template(_name, _start, _finish, _id, payload)
+          # retrieve the tracing context
+          tracing_context = payload.fetch(:tracing_context)
 
+          # create a new Span and add it to the tracing context
           tracer = ::Rails.configuration.datadog_trace.fetch(:tracer)
-          type = Datadog::Ext::HTTP::TEMPLATE
-          tracer.trace('rails.render_template', span_type: type)
+          span = tracer.trace('rails.render_template', span_type: Datadog::Ext::HTTP::TEMPLATE)
+          tracing_context[:dd_rails_template_span] = span
+        rescue StandardError => e
+          Datadog::Tracer.log.debug(e.message)
+        end
 
-          Thread.current[key] = true
+        def self.finish_render_template(_name, _start, _finish, _id, payload)
+          # retrieve the tracing context and the latest active span
+          tracing_context = payload.fetch(:tracing_context)
+          span = tracing_context[:dd_rails_template_span]
+          return unless span && !span.finished?
+
+          # finish the tracing and update the execution time
+          begin
+            tracer = ::Rails.configuration.datadog_trace.fetch(:tracer)
```

Comment:
Do we really need tracer here?

Reply:
ops, probably it was mostly because of the previous code, nice catch!